test/vm_tests/transformer/declaration_parsing_test.dart

```diff
@@ -140,6 +140,26 @@ main() {
           expect(declarations.declaresComponent, isTrue);
         });
 
+        test('a component with builder compatible annotated private props class', () {
+          setUpAndParse('''
+            @Factory()    UiFactory<FooProps> Foo;
+            class FooProps {}
```

Comment:
This class should follow the same setup that we will have in practice where it extends the private class and mixes in an accessors mixin.

```dart
class FooProps extends _$FooProps with _$FooPropsAccessorsMixin {}
```